### PR TITLE
Add in Concourse PR pipeline

### PR DESCRIPTION
**Pull request by cybojenix**
Based on https://github.com/cybojenix/govwifi-concourse-pr-template/tree/master/govwifi-pr

### CODE REVIEW

File: ci/tasks/scripts/pre-build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
```
Comment (emileswarts) [inline ci/tasks/scripts/pre-build.sh]: This need to be updated with set -e ... ?

Comment (cybojenix) [reply ci/tasks/scripts/pre-build.sh]: forgot to push. Done

File: ci/tasks/scripts/lint.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+set -e -u -o pipefail
+
+./src/ci/tasks/scripts/with-docker.sh
+
+workspace_dir="${PWD}"
+cd src
+
+make lint
+
+cd "${workspace_dir}"
```
Comment (emileswarts) [inline ci/tasks/scripts/lint.sh]: Pushd and popd could work nicely for this?

File: docker-compose.concourse.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# Overrides the `app` service to load from a prebuilt image when available
+
+services:
+  app:
+    # relates to ci/tasks/pre-build.yml
+    image: "app-prebuilt"
+    build:
+      cache_from:
+        - app-prebuilt
+
```
Comment (emileswarts) [inline docker-compose.concourse.yml]: Extra line here

File: ci/tasks/scripts/lint.sh (hunk 2 above)
Comment (cybojenix) [reply ci/tasks/scripts/lint.sh]: it adds additional overhead for reading the script.
If anything goes wrong, it goes wrong in unusual ways (usually due to the stack not being popped).

Let's just keep it simple

File: ci/variables.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+These variables are secrets, provided by the hosted Concourse, or ones we have set up
+
+### Github Access Token
+
+Stored under: `github-access-token`
+
+An access token for cloning public + private repos, and sending `status` updates to PRs.
+
+This is associated with the `govwifi-jenkins` Github user account.
```
Comment (emileswarts) [inline ci/variables.md]: Is there a task to create a github-concourse user account?

Comment (cybojenix) [reply ci/variables.md]: nope. does one need making? The current user account could be renamed to `govwifi-ci`

**Pull request merged**